### PR TITLE
PT-3751 full text search

### PR DESCRIPTION
**Pull request by alexbukata**

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -4633,9 +4638,26 @@ components:
         onSale:
           type: boolean
 
+    ItemSearchFullText:
+      type: object
+      properties:
+        text:
+          type: string
+        fields:
```
Comment (dmitry-saraykin) [inline openapi.yaml:4646]: if it is not required better to specify default

File: openapi.yaml [hunk 2]
```diff
@@ -4633,9 +4638,26 @@ components:
...
+              - NAME
+              - DESCRIPTION
+              - TRAIT_VALUE
+      required:
+        - text
+
     ItemsSearchSort:
       type: string
       enum:
+        - DEFAULT
```
Comment (dmitry-saraykin) [inline openapi.yaml:4660]: May be call it RELEVANCE?

File: openapi.yaml [hunk 3]
```diff
@@ -4563,15 +4563,20 @@ components:
           default: false
           example: "${filterDeleted}"
         names:
+          deprecated: true
+          description: "Deprecated. Please, use fullText instead"
```
Comment (dmitry-saraykin) [inline openapi.yaml:4567]: if this search doesn't work at all at the moment better to delete it without deprecation

Comment (alexbukata) [reply openapi.yaml:4567]: names is used, so deleted descriptions

File: openapi.yaml (hunk 2 above)
Comment (alexbukata) [reply openapi.yaml:4660]: done

File: openapi.yaml (hunk 1 above)
Comment (alexbukata) [reply openapi.yaml:4646]: done

**Review by dmitry-saraykin — APPROVED** [review]

**Pull request merged**